airflow/kubernetes/pod.py

```diff
@@ -33,45 +33,66 @@ class Resources(K8SModel):
     :type request_memory: str
     :param request_cpu: requested CPU number
     :type request_cpu: float | str
+    :param request_ephemeral_storage: requested ephermeral storage
+    :type request_ephemeral_storage: str
     :param limit_memory: limit for memory usage
     :type limit_memory: str
     :param limit_cpu: Limit for CPU used
     :type limit_cpu: float | str
     :param limit_gpu: Limits for GPU used
     :type limit_gpu: int
+    :param limit_ephemeral_storage: Limit for ephermeral storage
+    :type limit_ephemeral_storage: float | str
     """
     __slots__ = ('request_memory', 'request_cpu', 'limit_memory', 'limit_cpu', 'limit_gpu')
 
     def __init__(
             self,
             request_memory=None,
             request_cpu=None,
+            request_ephemeral_storage=None,
             limit_memory=None,
             limit_cpu=None,
-            limit_gpu=None):
+            limit_gpu=None,
+            limit_ephemeral_storage=None):
         self.request_memory = request_memory
         self.request_cpu = request_cpu
+        self.request_ephemeral_storage = request_ephemeral_storage
         self.limit_memory = limit_memory
         self.limit_cpu = limit_cpu
         self.limit_gpu = limit_gpu
+        self.limit_ephemeral_storage = limit_ephemeral_storage
 
     def is_empty_resource_request(self):
         """Whether resource is empty"""
         return not self.has_limits() and not self.has_requests()
 
     def has_limits(self):
         """Whether resource has limits"""
-        return self.limit_cpu is not None or self.limit_memory is not None or self.limit_gpu is not None
+        return self.limit_cpu is not None or \
+            self.limit_memory is not None or \
+            self.limit_gpu is not None or \
+            self.limit_ephemeral_storage is not None
 
     def has_requests(self):
         """Whether resource has requests"""
-        return self.request_cpu is not None or self.request_memory is not None
+        return self.request_cpu is not None or \
+            self.request_memory is not None or \
+            self.is_empty_resource_request is not None
```

Comment:
This line is wrong. I believe you're trying to reference `self.request_ephemeral_storage` and not `self.is_empty_resource_request` which would resolve always to True

Reply:
Thanks @konpap94 , will fix this